user/views.py

```diff
@@ -23,6 +24,50 @@ def form_valid(self, form):
       return redirect('home')
 
 
-class HomeView(LoginRequiredMixin, TemplateView):
+class HomeView(LoginRequiredMixin, ListView):
+  model = Tweet
+  queryset = Tweet.objects.order_by('published_at').reverse()
+  context_object_name = "tweet_list"
   template_name = 'user/home.html'
 
+
+class TweetView(LoginRequiredMixin, CreateView):
+  model = Tweet
+  fields = ['text']
+  template_name = "user/tweet.html"
+
+  def post(self, request):
+    self.form = TweetCreateForm(request.POST)
+    if self.form.is_valid():
+      tweet = self.form.save(commit=False)
+      tweet.user = request.user      
+      tweet.save()
+      return redirect('home')
+    else:
+      return redirect('tweet')
+
+
+
+class DeleteTweetView(UserPassesTestMixin, LoginRequiredMixin, DeleteView):
+  model = Tweet
+  success_url = reverse_lazy('home')
+
+  def test_func(self):
+    pk = self.kwargs["pk"]
+    tweet = Tweet.objects.get(pk=pk)
+    return tweet.user == self.request.user
```

Comment:
以下でできませんか
```suggestion
    return self.object.user == self.request.user
```

Reply:
<img width="561" alt="image" src="https://user-images.githubusercontent.com/83489580/183292242-5a35764b-5759-4c80-a656-838fda09dc30.png">
エラーが出てしまいました

Reply:
ごめんなさい！！！！！勘違いしてました！
以下でいけるはずです！
```suggestion
    tweet = self.get_object()
    return tweet.user == self.request.user
```